core/src/main/scala/kafka/utils/ReplicationUtils.scala

```diff
@@ -18,76 +18,39 @@
 package kafka.utils
 
 import kafka.api.LeaderAndIsr
-import kafka.controller.{IsrChangeNotificationHandler, LeaderIsrAndControllerEpoch}
-import kafka.utils.ZkUtils._
+import kafka.controller.LeaderIsrAndControllerEpoch
 import kafka.zk._
 import org.apache.kafka.common.TopicPartition
-import org.apache.zookeeper.data.Stat
-
-import scala.collection._
 
 object ReplicationUtils extends Logging {
 
-  private val IsrChangeNotificationPrefix = "isr_change_"
-
-  def updateLeaderAndIsr(zkClient: KafkaZkClient, topic: String, partitionId: Int, newLeaderAndIsr: LeaderAndIsr, controllerEpoch: Int,
-    zkVersion: Int): (Boolean,Int) = {
-    debug(s"Updated ISR for $topic-$partitionId to ${newLeaderAndIsr.isr.mkString(",")}")
-    val path = getTopicPartitionLeaderAndIsrPath(topic, partitionId)
-    val newLeaderData = LeaderAndIsrZNode.encode(newLeaderAndIsr, controllerEpoch)
+  def updateLeaderAndIsr(zkClient: KafkaZkClient, partition: TopicPartition, newLeaderAndIsr: LeaderAndIsr,
```

Comment:
I originally intended to move this to `KafkaZkClient` as well, but it seems to have slightly different behaviour than the batch version (`updateLeaderAndIsr`), so I thought it might be confusing. Feedback on the best way to handle that is welcome.

Reply:
Since this is only called by Partition, it seems we can leave it here.
When the second caller appears, we can decide a better place for this method.